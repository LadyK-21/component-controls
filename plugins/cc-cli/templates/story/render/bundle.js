rendered = doc.renderFn({ story, doc });