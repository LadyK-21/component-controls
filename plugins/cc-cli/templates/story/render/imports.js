rendered = renderExample({
  example,
  doc,
  config,
});