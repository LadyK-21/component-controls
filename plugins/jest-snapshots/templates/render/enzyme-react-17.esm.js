              const rendered = renderExample({
                example,
                doc,
                config,
              });
              const component = mount(rendered);
              const serialize = () => toJson(component, { mode: 'deep' });
