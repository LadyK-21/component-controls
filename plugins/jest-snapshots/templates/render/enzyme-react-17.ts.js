              let rendered;
{{=it.storeRender}}
              const component = mount(rendered);
              const serialize = () => toJson(component, { mode: 'deep' });
