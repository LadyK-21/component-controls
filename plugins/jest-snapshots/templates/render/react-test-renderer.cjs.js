              let rendered;
              renderer.act(() => {
                rendered = renderExample({
                  example,
                  doc,
                  config,
                });
              });
              const component = renderer.create(rendered);
              const serialize = component ? component.toJSON : () => undefined;
