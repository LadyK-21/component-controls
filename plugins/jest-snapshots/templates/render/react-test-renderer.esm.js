              let rendered;
              act(() => {
                rendered = renderExample({
                  example,
                  doc,
                  config,
                });
              });
              const component = rendered ? renderer.create(rendered) : undefined;
              const serialize = component ? component.toJSON : () => undefined;
