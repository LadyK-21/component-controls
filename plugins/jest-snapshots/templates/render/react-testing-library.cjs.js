              let rendered;
              act(() => {
                rendered = renderExample({
                  example,
                  doc,
                  config,
                });
              });
              const { asFragment: serialize } = render(rendered);
