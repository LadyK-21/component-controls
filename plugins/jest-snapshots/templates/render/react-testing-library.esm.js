              let rendered;
              act(() => {
                rendered = renderExample({
                  example,
                  doc,
                  config,
                });
              });
              let serialize;
              if (rendered) {
                ({ asFragment: serialize } = render(rendered));
              } else {
                serialize = () => 'error';
              }
