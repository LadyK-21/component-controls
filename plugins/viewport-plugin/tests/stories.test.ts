import path from 'path';
import MatchMediaMock from 'jest-matchmedia-mock';
import {
  loadConfigurations,
  extractDocuments,
} from '@component-controls/config';
import { renderExample } from '@component-controls/test-renderers';
import { render as reactRender } from '@component-controls/render/react';
import { render, act } from '@testing-library/react';

const renderErr = () => {
  throw new Error('Could not render the story');
};

describe('component-controls generated', () => {
  let matchMedia: MatchMediaMock;
  beforeAll(() => {
    jest.mock('rc-util/lib/Portal');
    matchMedia = new MatchMediaMock();
  });
  afterEach(() => {
    matchMedia.clear();
  });
  const configPath = path.resolve(__dirname, '../.config');
  const config = loadConfigurations(configPath);
  if (!config.renderFn) {
    config.renderFn = reactRender;
  }
  const documents = extractDocuments({ config, configPath });
  if (documents) {
    documents.forEach((file: string) => {
      const exports = require(file);
      const doc = exports.default;
      const examples = Object.keys(exports)
        .filter(key => key !== 'default')
        .map(key => exports[key]);

      if (examples.length) {
        describe(doc.title, () => {
          examples.forEach(example => {
            it(example.name, async () => {
              let rendered;
              act(() => {
                rendered = renderExample({
                  example,
                  doc,
                  config,
                });
              });
              if (!rendered) {
                renderErr();
                return;
              }
              const { asFragment } = render(rendered);
              expect(asFragment()).toMatchSnapshot();
            });
          });
        });
      }
    });
  }
});
