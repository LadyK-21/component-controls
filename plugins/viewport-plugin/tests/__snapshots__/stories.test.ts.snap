// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component-controls generated Plugins/ViewportBlock overview 1`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-jxxswp"
      id=""
    >
      <div
        class="css-xhp9da"
      >
        <h3
          class="css-16ist9v-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto"
      style="height: auto; overflow: visible;"
    >
      <div>
        <div>
          <div
            class="css-15owl46"
          >
            <div
              class="css-x60p33"
            >
              <div
                class="css-1vga9oy"
              >
                <div
                  style="display: inline-block;"
                >
                  <button
                    class="css-97huhz"
                  >
                    widths
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-eqxzio"
          >
            <div
              class="css-10nnqwz-ViewportBlock"
            >
              <div
                class="css-77fsv4-HoverBox"
              >
                <div
                  class="css-xtndwh"
                >
                  <div
                    class="css-bff87g-Story"
                  >
                    <div
                      class="css-jxxswp"
                      id=""
                    >
                      <div
                        class="css-xhp9da"
                      >
                        <h3
                          class="css-16ist9v-BlockContainer"
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="false"
                      class="rah-static rah-static--height-auto"
                      style="height: auto; overflow: visible;"
                    >
                      <div>
                        <div
                          class="css-154oin1"
                          data-testid="story"
                          id="blocks-core-story-plain--controls"
                        >
                          <div
                            class="story-render-container css-dhmrnb"
                          >
                            <svg
                              aria-valuemax="1"
                              aria-valuemin="0"
                              aria-valuenow="0.5"
                              class="css-14d2907"
                              fill="none"
                              height="128"
                              role="img"
                              stroke="currentcolor"
                              stroke-width="2"
                              viewBox="0 0 32 32"
                              width="128"
                            >
                              <circle
                                cx="16"
                                cy="16"
                                opacity="0.125"
                                r="14"
                              />
                              <circle
                                cx="16"
                                cy="16"
                                r="14"
                                stroke-dasharray="87.96459430051421"
                                stroke-dashoffset="43.982297150257104"
                                transform="rotate(-90 16 16)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  class="hoverbox-text css-1cgogy4"
                >
                  320px
                </span>
              </div>
              <div
                class="css-x0hg6m-HoverBox"
              >
                <div
                  class="css-xtndwh"
                >
                  <div
                    class="css-1xj1pl2-Story"
                  >
                    <div
                      class="css-jxxswp"
                      id=""
                    >
                      <div
                        class="css-xhp9da"
                      >
                        <h3
                          class="css-16ist9v-BlockContainer"
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="false"
                      class="rah-static rah-static--height-auto"
                      style="height: auto; overflow: visible;"
                    >
                      <div>
                        <div
                          class="css-154oin1"
                          data-testid="story"
                          id="blocks-core-story-plain--controls"
                        >
                          <div
                            class="story-render-container css-dhmrnb"
                          >
                            <svg
                              aria-valuemax="1"
                              aria-valuemin="0"
                              aria-valuenow="0.5"
                              class="css-14d2907"
                              fill="none"
                              height="128"
                              role="img"
                              stroke="currentcolor"
                              stroke-width="2"
                              viewBox="0 0 32 32"
                              width="128"
                            >
                              <circle
                                cx="16"
                                cy="16"
                                opacity="0.125"
                                r="14"
                              />
                              <circle
                                cx="16"
                                cy="16"
                                r="14"
                                stroke-dasharray="87.96459430051421"
                                stroke-dashoffset="43.982297150257104"
                                transform="rotate(-90 16 16)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  class="hoverbox-text css-1cgogy4"
                >
                  375px
                </span>
              </div>
              <div
                class="css-1j6jxrr-HoverBox"
              >
                <div
                  class="css-xtndwh"
                >
                  <div
                    class="css-1xpqjvh-Story"
                  >
                    <div
                      class="css-jxxswp"
                      id=""
                    >
                      <div
                        class="css-xhp9da"
                      >
                        <h3
                          class="css-16ist9v-BlockContainer"
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="false"
                      class="rah-static rah-static--height-auto"
                      style="height: auto; overflow: visible;"
                    >
                      <div>
                        <div
                          class="css-154oin1"
                          data-testid="story"
                          id="blocks-core-story-plain--controls"
                        >
                          <div
                            class="story-render-container css-dhmrnb"
                          >
                            <svg
                              aria-valuemax="1"
                              aria-valuemin="0"
                              aria-valuenow="0.5"
                              class="css-14d2907"
                              fill="none"
                              height="128"
                              role="img"
                              stroke="currentcolor"
                              stroke-width="2"
                              viewBox="0 0 32 32"
                              width="128"
                            >
                              <circle
                                cx="16"
                                cy="16"
                                opacity="0.125"
                                r="14"
                              />
                              <circle
                                cx="16"
                                cy="16"
                                r="14"
                                stroke-dasharray="87.96459430051421"
                                stroke-dashoffset="43.982297150257104"
                                transform="rotate(-90 16 16)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  class="hoverbox-text css-1cgogy4"
                >
                  768px
                </span>
              </div>
              <div
                class="css-aua95t-HoverBox"
              >
                <div
                  class="css-xtndwh"
                >
                  <div
                    class="css-xxu1iu-Story"
                  >
                    <div
                      class="css-jxxswp"
                      id=""
                    >
                      <div
                        class="css-xhp9da"
                      >
                        <h3
                          class="css-16ist9v-BlockContainer"
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="false"
                      class="rah-static rah-static--height-auto"
                      style="height: auto; overflow: visible;"
                    >
                      <div>
                        <div
                          class="css-154oin1"
                          data-testid="story"
                          id="blocks-core-story-plain--controls"
                        >
                          <div
                            class="story-render-container css-dhmrnb"
                          >
                            <svg
                              aria-valuemax="1"
                              aria-valuemin="0"
                              aria-valuenow="0.5"
                              class="css-14d2907"
                              fill="none"
                              height="128"
                              role="img"
                              stroke="currentcolor"
                              stroke-width="2"
                              viewBox="0 0 32 32"
                              width="128"
                            >
                              <circle
                                cx="16"
                                cy="16"
                                opacity="0.125"
                                r="14"
                              />
                              <circle
                                cx="16"
                                cy="16"
                                r="14"
                                stroke-dasharray="87.96459430051421"
                                stroke-dashoffset="43.982297150257104"
                                transform="rotate(-90 16 16)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  class="hoverbox-text css-1cgogy4"
                >
                  1024px
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`component-controls generated Plugins/ViewportBlock sizes 1`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-jxxswp"
      id=""
    >
      <div
        class="css-xhp9da"
      >
        <h3
          class="css-16ist9v-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto"
      style="height: auto; overflow: visible;"
    >
      <div>
        <div>
          <div
            class="css-15owl46"
          >
            <div
              class="css-x60p33"
            >
              <div
                class="css-1vga9oy"
              >
                <div
                  style="display: inline-block;"
                >
                  <button
                    class="css-97huhz"
                  >
                    widths
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-eqxzio"
          >
            <div
              class="css-10nnqwz-ViewportBlock"
            >
              <div
                class="css-1bxbzmw-HoverBox"
              >
                <div
                  class="css-xtndwh"
                >
                  <div
                    class="css-ks1xfj-Story"
                  >
                    <div
                      class="css-jxxswp"
                      id=""
                    >
                      <div
                        class="css-xhp9da"
                      >
                        <h3
                          class="css-16ist9v-BlockContainer"
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="false"
                      class="rah-static rah-static--height-auto"
                      style="height: auto; overflow: visible;"
                    >
                      <div>
                        <div
                          class="css-154oin1"
                          data-testid="story"
                          id="blocks-core-story-plain--controls"
                        >
                          <div
                            class="story-render-container css-dhmrnb"
                          >
                            <svg
                              aria-valuemax="1"
                              aria-valuemin="0"
                              aria-valuenow="0.5"
                              class="css-14d2907"
                              fill="none"
                              height="128"
                              role="img"
                              stroke="currentcolor"
                              stroke-width="2"
                              viewBox="0 0 32 32"
                              width="128"
                            >
                              <circle
                                cx="16"
                                cy="16"
                                opacity="0.125"
                                r="14"
                              />
                              <circle
                                cx="16"
                                cy="16"
                                r="14"
                                stroke-dasharray="87.96459430051421"
                                stroke-dashoffset="43.982297150257104"
                                transform="rotate(-90 16 16)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  class="hoverbox-text css-1cgogy4"
                >
                  xsmall
                </span>
              </div>
              <div
                class="css-y3p6ve-HoverBox"
              >
                <div
                  class="css-xtndwh"
                >
                  <div
                    class="css-1jzdp2u-Story"
                  >
                    <div
                      class="css-jxxswp"
                      id=""
                    >
                      <div
                        class="css-xhp9da"
                      >
                        <h3
                          class="css-16ist9v-BlockContainer"
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="false"
                      class="rah-static rah-static--height-auto"
                      style="height: auto; overflow: visible;"
                    >
                      <div>
                        <div
                          class="css-154oin1"
                          data-testid="story"
                          id="blocks-core-story-plain--controls"
                        >
                          <div
                            class="story-render-container css-dhmrnb"
                          >
                            <svg
                              aria-valuemax="1"
                              aria-valuemin="0"
                              aria-valuenow="0.5"
                              class="css-14d2907"
                              fill="none"
                              height="128"
                              role="img"
                              stroke="currentcolor"
                              stroke-width="2"
                              viewBox="0 0 32 32"
                              width="128"
                            >
                              <circle
                                cx="16"
                                cy="16"
                                opacity="0.125"
                                r="14"
                              />
                              <circle
                                cx="16"
                                cy="16"
                                r="14"
                                stroke-dasharray="87.96459430051421"
                                stroke-dashoffset="43.982297150257104"
                                transform="rotate(-90 16 16)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  class="hoverbox-text css-1cgogy4"
                >
                  small
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
